src/scale.c

```diff
@@ -0,0 +1,122 @@
+// Copyright 2021 Joe Drago. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "avif/internal.h"
+
+#if !defined(AVIF_LIBYUV_ENABLED)
+
+avifBool avifImageScale(avifImage * image, uint32_t dstWidth, uint32_t dstHeight, avifDiagnostics * diag)
+{
+    (void)image;
+    (void)dstWidth;
+    (void)dstHeight;
+    avifDiagnosticsPrintf(diag, "avifImageScale() called, but is unimplemented without libyuv!");
+    return AVIF_FALSE;
+}
+
+#else
+
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wstrict-prototypes" // "this function declaration is not a prototype"
+#endif
+#include <libyuv.h>
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
+
+// This should be configurable and/or smarter
+#define AVIF_LIBYUV_FILTER_MODE kFilterBox
```

Comment:
Add a comment to note that `kFilterBox` has the highest quality (but is also the slowest).

But actually, since the low-resolution frame will be displayed only temporarily, it seems better to use the fastest filter mode.

Reply:
I believe smooth version is preferred here. libjpeg-turbo even applies extra blur to smooth the decoded image: https://github.com/libjpeg-turbo/libjpeg-turbo/issues/459.